src/io/PImMSParser.m

```diff
@@ -0,0 +1,144 @@
+classdef PImMSParser < Parser
+    properties (Constant)
+        Name = 'PImMS Binary';
+    end
+
+    properties (SetAccess = protected)
+        % Time-of-flight properties
+        tof_start;
+        tof_end;
+        tof_range;
+
+        % Image data variable
+        image_data;
+    end
+
+    methods (Static)
+        function filterSpec = getFilterSpec()
+            filterSpec = {'*.bin', 'PImMS Binary (*.bin)'};
+        end
+    end
+
+    methods
+        function this = PImMSParser(filename)
+            this.filename = filename;
+
+            % Check that the filename ends in .bin
+            [pathstr,name,ext] = fileparts(filename);
+
+            if(~strcmpi(ext, '.bin'))
+                exception = MException('PImMSParser:FailedToParse', ...
+                    ['Must supply a BIN (*.bin) file']);
+                throw(exception);
+            end
+
+        end
+
+
+        function parse(this)
+            % Display a message to the user that parsing has started
+            notify(this, 'ParsingStarted');
+
+            % Assume data is PImMS2 (324x324)
+            % TODO: Perform any parsing of the data here. Making sure that width and height are filled in
+            this.width = 324;
+            this.height = 324;
```

Comment:
You have hardcoded values for the width and the height, can these be found in the data itself?

Reply:
The width and height are not explicitly included in the data, but there is the possibility that you could make two passes over the data, the first to find the maximum x, y and t coordinate to define the size of the arrays, and the second to actually read the data.